store/localstore/kv.go

```diff
@@ -58,14 +56,33 @@ func (s *dbStore) Seek(key []byte, startTS uint64) ([]byte, []byte, error) {
 			time.Sleep(time.Microsecond)
 			continue
 		}
-		// No conflict, safe to seek.
-		break
+		return nil
+	}
+}
+
+// Seek searches for the first key in the engine which is >= key in byte order, returns (nil, nil, ErrNotFound)
+// if such key is not found.
+func (s *dbStore) Seek(key []byte, startTS uint64) ([]byte, []byte, error) {
+	err := s.prepareSeek(startTS)
+	if err != nil {
+		return nil, nil, errors.Trace(err)
 	}
 	key, val, err := s.db.Seek(key)
 	s.wg.Done()
 	return key, val, err
 }
 
+// SeekReverse searches for the first key in the engine which is less than key in byte order.
+// Returns (nil, nil, ErrNotFound) if such key is not found.
+func (s *dbStore) SeekReverse(key []byte, startTS uint64) ([]byte, []byte, error) {
+	err := s.prepareSeek(startTS)
+	if err != nil {
+		return nil, nil, errors.Trace(err)
+	}
+	s.wg.Done()
+	return s.db.SeekReverse(key)
+}
```

Comment:
I think you want to `SeekReverse()` before `wg.Done()`, like the `Seek()` method above.